test/bb/Main.hs

```diff
@@ -54,11 +62,91 @@ main =
 -}
 bbFormRenames :: Property
 bbFormRenames = forAllShrinkShow (arbitrary @(DataDeclSet 'Poly1PolyThunks)) shrink prettyDeclSet $ \(DataDeclSet decls) ->
-  let bbfDecls = mapMaybe mkBBF decls -- only gives 'Nothing' if it's a 0 constructor type (a la `Void`) so no real point testing
-      results = mapM (runRenameM . renameValT) bbfDecls
-   in case results of -- all (isRight . runRenameM . renameValT) bbfDecls
-        Left err -> error (show err)
-        Right _ -> True
+  case traverse mkBBF decls of
+    Left _ -> False
+    Right (catMaybes -> bbfDecls) ->
+      let results =
+            mapM
+              ( \valT -> case runRenameM . renameValT $ valT of
+                  Left err -> Left (err, valT)
+                  Right res -> Right res
+              )
+              bbfDecls
+       in case results of -- all (isRight . runRenameM . renameValT) bbfDecls
+            Right {} -> True
+            Left err -> error (show err)
+
+bbfList :: TestTree
+bbfList = testCase "bbfList" $ do
+  let bbf = mkBBF list
+  bbf' <- either (assertFailure . show) (maybe (assertFailure "no bbf for list") pure) bbf
+  renamed <- either (assertFailure . show) pure . runRenameM . renameValT $ bbf'
```

Comment:
Why the renaming step? As far as I can tell, the tests are meant to ensure the form is correct: the renaming seems unnecessary.